tests/src/Unit/DataService/CachingContextFactoryTest.php

```diff
@@ -0,0 +1,205 @@
+<?php
```

Comment:
This was fun. I basically copied the test class from phpDoc, and proved I didn't break anything with the caching as the test didn't require any changes.